src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -488,20 +488,28 @@ static long long setLastFailure(const BSONObj& subOrig, CachedSubscription* subC
 
 /* ****************************************************************************
 *
-* setTimesFailed - 
+* setLastSuccess -
 */
-static void setTimesFailed(const BSONObj& subOrig, CachedSubscription* subCacheP, BSONObjBuilder* b)
+static long long setLastSuccess(const BSONObj& subOrig, CachedSubscription* subCacheP, BSONObjBuilder* b)
```

Comment:
Actually, lastNotification is added to before sending the notification while the other two (lastSuccess/lastFailure) need to wait until after to know the status of the notification and the resulting values are stored in the sub-cache only, to be added to mongo when a sub cache refresh is performed.  lastNotification works very differently.

I could change setLastNotification to return the value of lastNotification but that value is not used, so I really don't see any meaning with it ...

To be further discussed perhaps, but, in my opinion, NTC